modules/issue_tracker/php/NDB_Menu_Filter_Form_issue_tracker.class.inc

```diff
@@ -0,0 +1,384 @@
+    }
+
+
+    /**
+     * Overwrites the function to add a customized filter
+     * for watching and include closed
+     *
+     * @param string $prepared_key filter key
+     * @param string $field        filter field
+     * @param string $val          filter value
+     *
+     * @return null
+     */
+    function _addValidFilters($prepared_key, $field, $val)
+    {
+        $user  =& User::singleton();
+        $query = '';
+        if ((!empty($val) || $val === '0') && $field != 'order') {
+            if (($field != 'w.userID')
+                && ($field != 'minDate')
+                && ($field != 'maxDate')
+            ) {
+                if (in_array($field, $this->CheckboxFilters) && $val) {
+                    $query .= " AND $field";
+                } elseif (isset($this->EqualityFilters)
+                    && in_array($field, $this->EqualityFilters)
+                ) {
+                    $query .= " AND $field = :v_$prepared_key";
+                } else {
+                    $query .= " AND $field LIKE CONCAT('%', :v_$prepared_key, '%')";
+                }
+            } else if ($field = 'w.userID' && ($val == 1)) {
+                $query .= " AND w.userID = :v_$prepared_key";
+                //'" . $user->getData('UserID') . "'";
+            } else if ($field = 'minDate') {
+                $query .= " AND i.dateCreated > '" . $val . "00-00-00'";
+            } else if ($field = 'maxDate') {
+                $query .= " AND i.dateCreated < " . $val . "00-00-00'";
+            }
+        }
+        error_log($query);
+            return $query;
+    }
+
+
+    /**
+     * Utility function to re-organize query results for selects.
+     * Key and value are equal.
+     *
+     * @param array  $queryResult query result
+     * @param string $key         key needed to extract appropriate result column
+     *
+     * @return array
+     */
+    function reorganizeQueryResults($queryResult, $key)
+    {
+        $result = array('' => "All");
+        foreach ($queryResult as $r_row) {
+            $resultValue = $r_row[$key];
+            if ($resultValue) {
+                $result[$resultValue] = $resultValue;
+            }
+        }
+        return $result;
+    }
+
+    /**
+     * Converts this menu filter to an array of the form
+     *
+     * Headers => (string array)
+     * Data => (array of arrays of row data)
+     *
+     * @note overloaded function
+     *         Overloading this method to create a list of sessionID that
+     *         will be used for the Navigation Links in  the viewSession
+     *         page.
+     *
+     * @return associative array
+     */
+    function toArray()
+    {
+        $data  = parent::toArray();
+        $index = array_search('Issue ID', $data['Headers']);
+        if ($index !== false) {
+            $_SESSION['State']->setProperty(
+                'issuesListed',
+                array_column($data['Data'], $index)
+            );
+        }
```

Comment:
It provides the same functionality of toArray() that it overloads plus adds all of the issues that have been listed after applying filters to an array. This array is then added to the session state and used for navigation in the control panel. I copied this from imaging browser again.